File: Firebase/creative_api/functions/src/Service/DriverService.js
const Driver = require("./../Class/Driver");
const driverRepository = require("./../Repository/DriverRepository");

class DriverSevice {
    async saveDriver(driver) {
        return (await driverRepository.save(driver.get()));
    }
    async getById(id) {
        return (await driverRepository.find(id));
    }
    async getAll() {
        return (await driverRepository.findAll());
    }
    async updateDriver(driver){
        return (await driverRepository.update(driver.get()));
    }
    async findByUser(user) {
        return (await driverRepository.findByUser(user.getId()));
    }
}

module.exports = new DriverSevice();

File: classes PI/Truckleer/Truckleer/Modules/Driver/DriverService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Truckleer.Modules
{
    class DriverService
    {
        //Driver repository property
        readonly DriverRepository repository;
        //Class constructor
        public DriverService()
        {
            //initializate driver repository property
            repository = new DriverRepository();
        }
        //Method to findAll drivers
        public List<Driver> FindAll()
        {
            return repository.FindAll().Result;
        }
        //Method to find one specific driver by id
        public Driver FindOne(string Id)
        {
            return repository.Find(Id).Result;
        }
        //Method to Save or Update driver
        public bool Save(Driver driver)
        {
            return repository.Save(driver).Result;
        }
    }
}
